project/Build.scala

```diff
@@ -1265,17 +1265,13 @@ object Build {
         s"-source-links:github://lampepfl/dotty/$referenceVersion",
       ) ++ scalacOptionsDocSettings ++ revision ++ params ++ targets
       import _root_.scala.sys.process._
-      if (usingScript)
-        Def.task((s"$distLocation/bin/scaladoc" +: cmd).!)
-      else {
-        val escapedCmd = cmd.map(arg => if(arg.contains(" ")) s""""$arg"""" else arg)
-        Def.task {
-          try {
-            (Compile / run).toTask(escapedCmd.mkString(" ", " ", "")).value
-            0
-          } catch {
-            case _ : Throwable => 1
-          }
+      val escapedCmd = cmd.map(arg => if(arg.contains(" ")) s""""$arg"""" else arg)
+      Def.task {
+        try {
+          (Compile / run).toTask(escapedCmd.mkString(" ", " ", "")).value
+          0
+        } catch {
+          case _ : Throwable => 1
```

Comment:
Do you actually need to return 0 or 1 here instead of letting it throw an exception if it fails?

Reply:
Tbh it's the way it was when we ran part of tasks using script and the rest using runMain. Integer return type was there for compatibility between these two approaches but now it can be removed